opossum-tool/OpossumUI
# feat: Create SPDX document and add it to release

Create an SPDX document for the project and add it to the release

### Summary of changes

- Add the Yarn plugin [@yarnpkg/plugin-spdx](https://github.com/spdx/yarn-plugin-spdx)
- Use the plugin in CI/CD workflow `publish-release` to generate an SPDX document for the project and add it to the release
- Use the plugin in CI/CD workflow `build-and-e2e-test`

### Context and reason for change

It is desirable for the OpossumUI releases to contain an SBOM document.

### How can the changes be tested

Run the command `yarn spdx` verify that an SPDX document with the name opossum-ui.spdx.json is created in the root of the project and inspect it.
To visualize the generated document, the [SPDX Python tooling](https://github.com/spdx/tools-python) can be used. To do so, use the command pyspdxtools -I <local path to generated spdx document> --graph -o <output path>.
![opossum-ui](https://github.com/opossum-tool/OpossumUI/assets/48415604/cb6a736f-8638-427f-bbb8-fb4295a97d3e)

### Note

The generated SPDX document is currently invalid because the dependency `babel-plugin-transform-import-meta` has the license `BSD` declared, which is not in the [SPDX license list](https://spdx.org/licenses/).
There is a ticket for this in the plugin implementation: https://github.com/spdx/yarn-plugin-spdx/issues/12
We should wait for this to be fixed before landing this PR.

Note: Please review the [guidelines for contributing](https://github.com/opossum-tool/OpossumUI/blob/main/CONTRIBUTING.md) to this repository.
